static/jsonschema/kedro-catalog-0.18.0.json

```
@@ -0,0 +1,1042 @@
+          "then": {
+            "required": ["dataset"],
+            "properties": {
+              "dataset": {
+                "pattern": ".*",
+                "description": "A Kedro DataSet object or a dictionary to cache."
+              },
+              "copy_mode": {
+                "type": "string",
+                "description": "The copy mode used to copy the data. Possible\nvalues are: \"deepcopy\", \"copy\" and \"assign\". If not\nprovided, it is inferred based on the data type."
+              }
+            }
+          }
+        },
+        {
+          "if": { "properties": { "type": { "const": "IncrementalDataSet" } } },
+          "then": {
+            "required": ["path", "dataset"],
+            "properties": {
+              "path": {
+                "type": "string",
+                "description": "Path to the folder containing partitioned data.\nIf path starts with the protocol (e.g., ``s3://``) then the\ncorresponding ``fsspec`` concrete filesystem implementation will\nbe used. If protocol is not specified,\n``fsspec.implementations.local.LocalFileSystem`` will be used.\n**Note:** Some concrete implementations are bundled with ``fsspec``,\nwhile others (like ``s3`` or ``gcs``) must be installed separately\nprior to usage of the ``PartitionedDataSet``."
+              },
+              "dataset": {
+                "pattern": ".*",
+                "description": "Underlying dataset definition. This is used to instantiate\nthe dataset for each file located inside the ``path``.\nAccepted formats are:\na) object of a class that inherits from ``AbstractDataSet``\nb) a string representing a fully qualified class name to such class\nc) a dictionary with ``type`` key pointing to a string from b),\nother keys are passed to the Dataset initializer.\nCredentials for the dataset can be explicitly specified in\nthis configuration."
+              },
+              "checkpoint": {
+                "pattern": "object",
+                "description": "Optional checkpoint configuration. Accepts a dictionary\nwith the corresponding dataset definition including ``filepath``\n(unlike ``dataset`` argument). Checkpoint configuration is\ndescribed here:\nhttps://kedro.readthedocs.io/en/stable/05_data/02_kedro_io.html#checkpoint-configuration\nCredentials for the checkpoint can be explicitly specified\nin this configuration."
+              },
+              "filepath_arg": {
+                "type": "string",
+                "description": "Underlying dataset initializer argument that will\ncontain a path to each corresponding partition file.\nIf unspecified, defaults to \"filepath\"."
+              },
+              "filename_suffix": {
+                "type": "string",
+                "description": "If specified, only partitions that end with this\nstring will be processed."
+              },
+              "credentials": {
+                "type": "object",
```

This is super off topic - but we also have this change on our backlog so it would be great to add in here:
```suggestion
                "type": ["object", "string"],
```

---

I can make those changes to all datasets if you want

---

@datajoely should we make this change when it's relevant, if it is not implemented yet? I'd rather leave it out for now.